### DISCUSSION TITLE
Newest Version Order

### DISCUSSION THREAD

**Discussion by RAOEUS** [microsoft/winget-pkgs#168706] [Q&A] [answered]
I recently added a new version of `OpenSteoProject.Plover` to the repo, 4.0.0rc2, but winget still sees 4.0.0.dev12 as the latest version. Can I force it to see 4.0.0rc2 as the latest version through the config, or must I rename the files? [Here is the manifest URL.](https://github.com/microsoft/winget-pkgs/tree/master/manifests/o/OpenStenoProject/Plover)

Thank you for your time!

**Comment by RAOEUS** [comment]
![image](https://github.com/user-attachments/assets/50c3e8c8-57d0-4294-978b-c12837651997)
You can see here that I have 4.0.0rc2 installed, but it says 4.0.0.dev12 is the most recent version. This is incorrect.

**Comment by Trenly** [comment, marked as answer]
Take a look at [how versions are parsed in WinGet](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#version-sorting-in-winget). It's the extra `.` thats causing the issue

Breaking it down, the versions are compared like this

| Part A | Part B | Comparison |
| --- | --- | --- |
| 4 | 4 | Equal |
| 0 | 0 | Equal |
| 0rc2 | 0 | B is greater since it has no trailing alpha- characters |
| 0 | dev12 | Not Compared |

To fix this, `4.0.0rc2` could be moved to `4.0.0.rc2`; *or* `4.0.0.dev12` could be moved to `4.0.0dev12`
